x/wasm/keeper/keeper.go

```diff
@@ -482,26 +484,17 @@ func (k Keeper) migrate(
 		contractInfo.IBCPortID = ibcPort
 	}
 
-	env := types.NewEnv(sdkCtx, contractAddress)
-
-	// prepare querier
-	querier := k.newQueryHandler(sdkCtx, contractAddress)
-
-	prefixStoreKey := types.GetContractStorePrefix(contractAddress)
-	vmStore := types.NewStoreAdapter(prefix.NewStore(runtime.KVStoreAdapter(k.storeService.OpenKVStore(sdkCtx)), prefixStoreKey))
-	gasLeft := k.runtimeGasForContract(sdkCtx)
-	res, gasUsed, err := k.wasmVM.Migrate(newCodeInfo.CodeHash, env, msg, vmStore, cosmwasmAPI, &querier, k.gasMeter(sdkCtx), gasLeft, costJSONDeserialization)
-	k.consumeRuntimeGas(sdkCtx, gasUsed)
-	if err != nil {
-		return nil, errorsmod.Wrap(types.ErrVMError, err.Error())
-	}
-	if res == nil {
-		// If this gets executed, that's a bug in wasmvm
-		return nil, errorsmod.Wrap(types.ErrVMError, "internal wasmvm error")
-	}
-	if res.Err != "" {
-		return nil, types.MarkErrorDeterministic(errorsmod.Wrap(types.ErrMigrationFailed, res.Err))
+	// check for migrate endpoint
+	hasMigrateEndpoint := false
+	switch {
+	case contains(report.Entrypoints, "migrate") && msg != nil:
+		hasMigrateEndpoint = true
+	case !contains(report.Entrypoints, "migrate") && msg == nil:
+		hasMigrateEndpoint = false
```

Comment:
- Wouldn't the flow become easier if we set `hasMigrateEndpoint := contains(report.Entrypoints, "migrate")` first and then check for valid combinations? No need to execute the `contains` twice.
- I would use `len(msg) == 0` for the message check since both nil and an empty byte slice means there is no message set